### ISSUE TITLE
Compilation error on osx-arm64 with 0.3.14

### ISSUE THREAD

**Issue by h-vetinari** [xianyi/OpenBLAS#3153]
Not sure if this is specific to the setup on conda-forge (e.g. on clang 11.1 already), but it seems that https://github.com/xianyi/OpenBLAS/commit/34753eaebb8b2ddbc256e9e996c1fb315396a2a0 [breaks](https://github.com/conda-forge/openblas-feedstock/pull/116#issuecomment-801466264) the compilation for osx-arm64:
```
In file included from getarch_2nd.c:7:
In file included from ./common.h:457:
./common_arm64.h:69:38: error: unknown register name 'x2' in asm
                         : "memory", "x2" , "x3", "x4"
                                     ^
1 error generated.
make: *** [Makefile.prebuild:74: getarch_2nd] Error 1
```

[Reverting](https://github.com/conda-forge/openblas-feedstock/pull/116/commits/659f9c0c3a73a4c8c1501b5b4cb9a9c8e780dfba) that commit [results](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=291744&view=results) in successful compilation.

**Comment by martin-frbg** [comment]
Sorry for that. (Stumbled on something related when I tried to prepare Windows binaries). Should be fixed now with #3154

**Comment by h-vetinari** [comment]
Thanks for the quick handling!
